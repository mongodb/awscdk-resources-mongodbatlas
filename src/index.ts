// Copyright 2023 MongoDB Inc
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// L1 Constructors

export {
  CfnAccessListApiKey,
  CfnAccessListApiKeyProps,
} from "./l1-resources/access-list-api-key";

export {
  CfnAlertConfiguration,
  CfnAlertConfigurationProps,
  IntegerThresholdView,
  IntegerThresholdViewOperator,
  Matcher,
  MatcherFieldName,
  MatcherOperator,
  MetricThresholdView,
  MetricThresholdViewMode,
  MetricThresholdViewOperator,
  NotificationView,
  NotificationViewDatadogRegion,
  NotificationViewTypeName,
  NotificationViewOpsGenieRegion,
  NotificationViewRoles,
  NotificationViewSeverity,
} from "./l1-resources/alert-configuration";

export {
  CfnApiKey,
  CfnApiKeyProps,
  ProjectAssignment,
} from "./l1-resources/api-key";

export { CfnAuditing, CfnAuditingProps } from "./l1-resources/auditing";

export {
  CfnCloudBackUpRestoreJobs,
  CfnCloudBackUpRestoreJobsProps,
  CfnCloudBackUpRestoreJobsPropsLinks,
  CfnCloudBackUpRestoreJobsPropsDeliveryType,
  CfnCloudBackUpRestoreJobsPropsInstanceType,
  SynchronousCreationOptions,
} from "./l1-resources/cloud-backup-restore-jobs";

export {
  ApiAtlasDiskBackupCopySettingView,
  ApiDeleteCopiedBackupsView,
  ApiPolicyItemView,
  ApiPolicyView,
  CfnCloudBackupSchedule,
  CfnCloudBackupScheduleProps,
  Export,
  Link,
} from "./l1-resources/cloud-backup-schedule";

export {
  ApiAtlasDiskBackupShardedClusterSnapshotMemberView,
  ApiAtlasDiskBackupShardedClusterSnapshotMemberViewCloudProvider,
  ApiAtlasDiskBackupShardedClusterSnapshotView,
  ApiAtlasDiskBackupShardedClusterSnapshotViewFrequencyType,
  ApiAtlasDiskBackupShardedClusterSnapshotViewSnapshotType,
  ApiAtlasDiskBackupShardedClusterSnapshotViewStatus,
  ApiAtlasDiskBackupShardedClusterSnapshotViewType,
  CfnCloudBackupSnapshot,
  CfnCloudBackupSnapshotProps,
  CfnCloudBackupSnapshotPropsFrequencyType,
  CfnCloudBackupSnapshotPropsSnapshotType,
  CfnCloudBackupSnapshotPropsInstanceType,
} from "./l1-resources/cloud-backup-snapshot";

export {
  CfnCloudBackupSnapshotExportBucket,
  CfnCloudBackupSnapshotExportBucketProps,
} from "./l1-resources/cloud-backup-snapshot-export-bucket";

export {
  CfnClusterOutageSimulation,
  CfnClusterOutageSimulationProps,
  Filter,
  FilterCloudProvider,
} from "./l1-resources/cluster-outage-simulation";

export {
  CfnClusterProps,
  ProcessArgs,
  CfnClusterPropsBiConnector,
  CfnClusterPropsLabels,
  ConnectionStrings,
  AdvancedReplicationSpec,
  AdvancedAutoScaling,
  Specs,
  DiskGb,
  Compute,
  AdvancedRegionConfig,
  CfnCluster,
  CfnClusterPropsEncryptionAtRestProvider,
  AdvancedRegionConfigProviderName,
  Tag,
} from "./l1-resources/cluster";

export {
  Action,
  CfnCustomDbRole,
  CfnCustomDbRoleProps,
  InheritedRole,
  Resource,
} from "./l1-resources/custom-db-role";

export {
  CfnCustomDnsConfigurationClusterAws,
  CfnCustomDnsConfigurationClusterAwsProps,
} from "./l1-resources/custom-dns-configuration-cluster-aws";

export {
  CfnDataLakePipeline,
  CfnDataLakePipelineProps,
  Sink,
  SinkType,
  Source,
  SourceType,
  Transformations,
  PartitionFields,
} from "./l1-resources/data-lake-pipeline";

export {
  CfnDatabaseUser,
  CfnDatabaseUserProps,
  LabelDefinition,
  RoleDefinition,
  ScopeDefinition,
  CfnDatabaseUserPropsX509Type,
  CfnDatabaseUserPropsAwsiamType,
  CfnDatabaseUserPropsLdapAuthType,
  ScopeDefinitionType,
} from "./l1-resources/database-user";

export {
  CfnDataLakes,
  CfnDataLakesProps,
  DataLakeAwsCloudProviderConfigView,
  DataLakeCloudProviderConfigView,
  DataLakeDataProcessRegionView,
  DataLakeDataProcessRegionViewCloudProvider,
  DataLakeDataProcessRegionViewRegion,
  DataLakeDatabaseCollectionView,
  DataLakeDatabaseDataSourceView,
  DataLakeDatabaseDataSourceViewDefaultFormat,
  DataLakeDatabaseView,
  DataLakeStorageView,
  DataLakeViewView,
  StoreDetail,
} from "./l1-resources/datalakes";

export {
  AwsKmsConfig,
  CfnEncryptionAtRest,
  CfnEncryptionAtRestProps,
} from "./l1-resources/encryption-at-rest";

export {
  CfnFederatedDatabaseInstance,
  CfnFederatedDatabaseInstanceProps,
  CloudProviderConfig,
  DataProcessRegion,
  Storage,
  Database,
  Store,
  Collection,
  DataSource,
  ReadPreference,
  TagSet,
  View,
} from "./l1-resources/federated-database-instance";

export {
  CfnFederatedQueryLimit,
  CfnFederatedQueryLimitProps,
  CfnFederatedQueryLimitPropsLimitName,
} from "./l1-resources/federated-query-limit";

export {
  CfnFederatedSettingsOrgRoleMapping,
  CfnFederatedSettingsOrgRoleMappingProps,
  RoleAssignment,
} from "./l1-resources/federated-settings-org-role-mapping";

export {
  CfnGlobalClusterConfig,
  CfnGlobalClusterConfigProps,
  ManagedNamespace,
  ZoneMapping,
} from "./l1-resources/global-cluster-config";

export {
  ApiAtlasNdsUserToDnMappingView,
  CfnLdapConfiguration,
  CfnLdapConfigurationProps,
} from "./l1-resources/ldap-configuration";

export {
  CfnLdapVerify,
  CfnLdapVerifyProps,
  Validation,
} from "./l1-resources/ldap-verify";

export {
  CfnMaintenanceWindow,
  CfnMaintenanceWindowProps,
} from "./l1-resources/maintenance-window";

export {
  CfnNetworkContainer,
  CfnNetworkContainerProps,
} from "./l1-resources/network-container";

export {
  CfnNetworkPeering,
  CfnNetworkPeeringProps,
} from "./l1-resources/network-peering";

export {
  CfnOnlineArchive,
  CfnOnlineArchiveProps,
  CfnOnlineArchivePropsCollectionType,
  CriteriaView,
  CriteriaViewDateFormat,
  CriteriaViewType,
  PartitionFieldView,
  ScheduleView,
  ScheduleViewType,
} from "./l1-resources/online-archive";

export {
  CfnOrgInvitation,
  CfnOrgInvitationProps,
  CfnOrgInvitationPropsRoles,
} from "./l1-resources/org-invitation";

export {
  CfnOrganization,
  CfnOrganizationProps,
  ApiKey,
} from "./l1-resources/organization";

export {
  CfnPrivateEndpointProps,
  CfnPrivateEndpoint,
  PrivateEndpoint,
} from "./l1-resources/private-endpoint";

export {
  CfnPrivateEndpointAdl,
  CfnPrivateEndpointAdlProps,
} from "./l1-resources/private-endpoint-adl";

export {
  CfnPrivateEndpointAwsProps,
  CfnPrivateEndpointAws,
} from "./l1-resources/private-endpoint-aws";

export {
  CfnPrivateEndPointRegionalMode,
  CfnPrivateEndPointRegionalModeProps,
} from "./l1-resources/private-endpoint-regional-mode";

export {
  CfnPrivateEndpointService,
  CfnPrivateEndpointServiceProps,
  CfnPrivateEndpointServicePropsCloudProvider,
} from "./l1-resources/private-endpoint-service";

export {
  CfnPrivatelinkEndpointServiceDataFederationOnlineArchive,
  CfnPrivatelinkEndpointServiceDataFederationOnlineArchiveProps,
} from "./l1-resources/privatelink-endpoint-service-data-federation-online-archive";

export {
  CfnProject,
  CfnProjectProps,
  ProjectApiKey,
  ProjectSettings,
  ProjectTeam,
} from "./l1-resources/project";

export {
  CfnProjectInvitation,
  CfnProjectInvitationProps,
  CfnProjectInvitationPropsRoles,
} from "./l1-resources/project-invitation";

export {
  AccessListDefinition,
  CfnProjectIpAccessList,
  CfnProjectIpAccessListProps,
  ListOptions,
} from "./l1-resources/project-ip-access-list";

export {
  CfnSearchDeployment,
  CfnSearchDeploymentProps,
  ApiSearchDeploymentSpec,
} from "./l1-resources/search-deployment";

export {
  ApiAtlasFtsAnalyzersTokenizer,
  ApiAtlasFtsAnalyzersViewManual,
  ApiAtlasFtsMappingsViewManual,
  ApiAtlasFtsSynonymMappingDefinitionView,
  CfnSearchIndex,
  CfnSearchIndexProps,
  SynonymSource,
} from "./l1-resources/search-index";

export {
  CfnServerlessInstance,
  CfnServerlessInstanceProps,
  ServerlessInstanceConnectionStrings,
  ServerlessInstancePrivateEndpoint,
  ServerlessInstancePrivateEndpointEndpoint,
  ServerlessInstancePrivateEndpointType,
  ServerlessInstanceProviderSettings,
  ServerlessInstanceProviderSettingsProviderName,
} from "./l1-resources/serverless-instance";

export {
  CfnServerlessPrivateEndpointProps,
  CfnServerlessPrivateEndpoint,
  AwsPrivateEndpointConfig,
} from "./l1-resources/serverless-private-endpoint";

export {
  CfnStreamConnection,
  CfnStreamConnectionProps,
  CfnStreamConnectionPropsType,
  DbRoleToExecute,
  DbRoleToExecuteType,
  StreamsKafkaAuthentication,
  StreamsKafkaSecurity,
} from "./l1-resources/stream-connection";

export {
  CfnStreamInstance,
  StreamsDataProcessRegionCloudProvider,
  CfnStreamInstanceProps,
  StreamsDataProcessRegion,
  StreamConfig,
  StreamsConnection,
  StreamsConnectionType,
} from "./l1-resources/stream-instance";

export {
  CfnTeams,
  CfnTeamsProps,
  CfnTeamsPropsRoleNames,
  AtlasRoleRoleName,
  AtlasRole,
  AtlasUser,
} from "./l1-resources/teams";

export {
  CfnThirdPartyIntegration,
  CfnThirdPartyIntegrationProps,
  CfnThirdPartyIntegrationPropsScheme,
  CfnThirdPartyIntegrationPropsServiceDiscovery,
  CfnThirdPartyIntegrationPropsType,
} from "./l1-resources/third-party-integration";

export {
  AuthConfig,
  AuthConfigOperationType,
  AuthConfigProviders,
  CfnTrigger,
  CfnTriggerProps,
  DatabaseConfig,
  DatabaseConfigOperationTypes,
  Event,
  EventAwseventbridge,
  EventAwseventbridgeAwsConfig,
  EventFunction,
  EventFunctionFuncConfig,
  ScheduleConfig,
} from "./l1-resources/trigger";

export {
  CfnX509AuthenticationDatabaseUser,
  CfnX509AuthenticationDatabaseUserProps,
  CustomerX509,
} from "./l1-resources/x509-authentication-database-user";

// L2 Constructors

export * from "./l2-resources/encryption-at-rest";
export * from "./l2-resources/third-party-integration";

// L3 Constructors

export * from "./l3-resources/common/props";

export * from "./l3-resources/atlas-basic";
export * from "./l3-resources/atlas-serverless-basic";
export * from "./l3-resources/atlas-basic-private-endpoint";
export * from "./l3-resources/encryption-at-rest-express";
export * from "./l3-resources/atlas-bootstrap";
