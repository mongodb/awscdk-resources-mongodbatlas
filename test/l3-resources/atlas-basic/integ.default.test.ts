import { App, Stack, aws_ec2 as ec2 } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import * as atlas from "../../../src";

const app = new App();

const stack = new Stack(app, "atlas-basic-default", {
  env: {
    region: process.env.CDK_DEFAULT_REGION,
    account: process.env.CDK_DEFAULT_ACCOUNT,
  },
});

test("should match snapshots", () => {
  const cluster = new atlas.AtlasBasic(stack, "atlas-basic-cluster", {
    orgId: "mock",
    accessList: [{ ipAddress: "0.0.0.0/0", comment: "public to all" }],
    profile: "my-mongo-profile",
  });

  const vpc = new ec2.Vpc(stack, "Vpc", { natGateways: 1 });

  cluster.addPrivateEndpoint({
    vpc,
    vpcSubnets: [
      { subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS, onePerAz: true },
    ],
  });

  const t = Template.fromStack(stack);
  expect(t).toMatchSnapshot();
});
