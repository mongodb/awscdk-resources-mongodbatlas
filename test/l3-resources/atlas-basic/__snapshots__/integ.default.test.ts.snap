// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "Vpc8378EB38": Object {
      "Properties": Object {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "atlas-basic-default/Vpc",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "VpcIGWD7BA715C": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "atlas-basic-default/Vpc",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "VpcPrivateSubnet1DefaultRouteBE02A9ED": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "VpcPublicSubnet1NATGateway4D7517AA",
        },
        "RouteTableId": Object {
          "Ref": "VpcPrivateSubnet1RouteTableB2C5B500",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcPrivateSubnet1RouteTableAssociation70C59FA6": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "VpcPrivateSubnet1RouteTableB2C5B500",
        },
        "SubnetId": Object {
          "Ref": "VpcPrivateSubnet1Subnet536B997A",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcPrivateSubnet1RouteTableB2C5B500": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "atlas-basic-default/Vpc/PrivateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcPrivateSubnet1Subnet536B997A": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.128.0/18",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "atlas-basic-default/Vpc/PrivateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcPrivateSubnet2DefaultRoute060D2087": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "VpcPublicSubnet1NATGateway4D7517AA",
        },
        "RouteTableId": Object {
          "Ref": "VpcPrivateSubnet2RouteTableA678073B",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcPrivateSubnet2RouteTableA678073B": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "atlas-basic-default/Vpc/PrivateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcPrivateSubnet2RouteTableAssociationA89CAD56": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "VpcPrivateSubnet2RouteTableA678073B",
        },
        "SubnetId": Object {
          "Ref": "VpcPrivateSubnet2Subnet3788AAA1",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcPrivateSubnet2Subnet3788AAA1": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.192.0/18",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "atlas-basic-default/Vpc/PrivateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcPublicSubnet1DefaultRoute3DA9E72A": Object {
      "DependsOn": Array [
        "VpcVPCGWBF912B6E",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "VpcIGWD7BA715C",
        },
        "RouteTableId": Object {
          "Ref": "VpcPublicSubnet1RouteTable6C95E38E",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcPublicSubnet1EIPD7E02669": Object {
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "atlas-basic-default/Vpc/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "VpcPublicSubnet1NATGateway4D7517AA": Object {
      "DependsOn": Array [
        "VpcPublicSubnet1DefaultRoute3DA9E72A",
        "VpcPublicSubnet1RouteTableAssociation97140677",
      ],
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "VpcPublicSubnet1EIPD7E02669",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "VpcPublicSubnet1Subnet5C2D37C4",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "atlas-basic-default/Vpc/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "VpcPublicSubnet1RouteTable6C95E38E": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "atlas-basic-default/Vpc/PublicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcPublicSubnet1RouteTableAssociation97140677": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "VpcPublicSubnet1RouteTable6C95E38E",
        },
        "SubnetId": Object {
          "Ref": "VpcPublicSubnet1Subnet5C2D37C4",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcPublicSubnet1Subnet5C2D37C4": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.0.0/18",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "atlas-basic-default/Vpc/PublicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcPublicSubnet2DefaultRoute97F91067": Object {
      "DependsOn": Array [
        "VpcVPCGWBF912B6E",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "VpcIGWD7BA715C",
        },
        "RouteTableId": Object {
          "Ref": "VpcPublicSubnet2RouteTable94F7E489",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcPublicSubnet2RouteTable94F7E489": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "atlas-basic-default/Vpc/PublicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcPublicSubnet2RouteTableAssociationDD5762D8": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "VpcPublicSubnet2RouteTable94F7E489",
        },
        "SubnetId": Object {
          "Ref": "VpcPublicSubnet2Subnet691E08A3",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcPublicSubnet2Subnet691E08A3": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.64.0/18",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "atlas-basic-default/Vpc/PublicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcVPCGWBF912B6E": Object {
      "Properties": Object {
        "InternetGatewayId": Object {
          "Ref": "VpcIGWD7BA715C",
        },
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
    "atlasbasicclusteraccessListatlasbasiccluster8366D58F": Object {
      "Properties": Object {
        "AccessList": Array [
          Object {
            "Comment": "public to all",
            "IPAddress": "0.0.0.0/0",
          },
        ],
        "Profile": "my-mongo-profile",
        "ProjectId": Object {
          "Fn::GetAtt": Array [
            "atlasbasicclusterprojectatlasbasiccluster86AB6E9F",
            "Id",
          ],
        },
      },
      "Type": "MongoDB::Atlas::ProjectIpAccessList",
    },
    "atlasbasicclusterclusteratlasbasiccluster56BD6853": Object {
      "Properties": Object {
        "ClusterType": "REPLICASET",
        "Name": "clusterclusteratlas-basic-cluster",
        "Profile": "my-mongo-profile",
        "ProjectId": Object {
          "Fn::GetAtt": Array [
            "atlasbasicclusterprojectatlasbasiccluster86AB6E9F",
            "Id",
          ],
        },
      },
      "Type": "MongoDB::Atlas::Cluster",
    },
    "atlasbasicclusterprivateendpointatlasbasiccluster0F4411EB": Object {
      "Properties": Object {
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "atlasbasicclusterprojectatlasbasiccluster86AB6E9F",
            "Id",
          ],
        },
        "PrivateEndpoints": Array [
          Object {
            "SubnetIds": Array [
              Object {
                "Ref": "VpcPrivateSubnet1Subnet536B997A",
              },
              Object {
                "Ref": "VpcPrivateSubnet2Subnet3788AAA1",
              },
            ],
            "VpcId": Object {
              "Ref": "Vpc8378EB38",
            },
          },
        ],
        "Profile": "my-mongo-profile",
        "Region": "us-east-1",
      },
      "Type": "MongoDB::Atlas::PrivateEndpoint",
    },
    "atlasbasicclusterprojectatlasbasiccluster86AB6E9F": Object {
      "Properties": Object {
        "Name": "projectprojectatlas-basic-cluster",
        "OrgId": "mock",
        "Profile": "my-mongo-profile",
      },
      "Type": "MongoDB::Atlas::Project",
    },
    "atlasbasicclusteruseratlasbasicclusterBD57D542": Object {
      "Properties": Object {
        "DatabaseName": "admin",
        "Password": "atlas-pwd",
        "Profile": "my-mongo-profile",
        "ProjectId": Object {
          "Fn::GetAtt": Array [
            "atlasbasicclusterprojectatlasbasiccluster86AB6E9F",
            "Id",
          ],
        },
        "Roles": Array [
          Object {
            "DatabaseName": "admin",
            "RoleName": "atlasAdmin",
          },
        ],
        "Username": "atlas-user",
      },
      "Type": "MongoDB::Atlas::DatabaseUser",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
